### ISSUE TITLE
Memory Read Crash with Firebase Performance Monitoring

### ISSUE THREAD

**Issue by wimbledon** [AFNetworking/AFNetworking#4152]
Environment:
iOS 10.3, 11.2.5 on simulators and iPhones
Xcode 8.3.3

Podfile.lock:
```
- AFNetworking (3.2.0):
    - AFNetworking/NSURLSession (= 3.2.0)
    - AFNetworking/Reachability (= 3.2.0)
    - AFNetworking/Security (= 3.2.0)
    - AFNetworking/Serialization (= 3.2.0)
    - AFNetworking/UIKit (= 3.2.0)
  - AFNetworking/NSURLSession (3.2.0):
    - AFNetworking/Reachability
    - AFNetworking/Security
    - AFNetworking/Serialization
  - AFNetworking/Reachability (3.2.0)
  - AFNetworking/Security (3.2.0)
  - AFNetworking/Serialization (3.2.0)
  - AFNetworking/UIKit (3.2.0):
    - AFNetworking/NSURLSession
  - Firebase/Core (4.8.2):
    - FirebaseAnalytics (= 4.0.9)
    - FirebaseCore (= 4.0.14)
  - Firebase/Performance (4.8.2):
    - Firebase/Core
    - FirebasePerformance (= 1.1.1)
  - FirebaseAnalytics (4.0.9):
    - FirebaseCore (~> 4.0)
    - FirebaseInstanceID (~> 2.0)
    - GoogleToolboxForMac/NSData+zlib (~> 2.1)
    - nanopb (~> 0.3)
  - FirebaseCore (4.0.14):
    - GoogleToolboxForMac/NSData+zlib (~> 2.1)
  - FirebaseInstanceID (2.0.8):
    - FirebaseCore (~> 4.0)
  - FirebasePerformance (1.1.1):
    - FirebaseAnalytics (~> 4.0)
    - FirebaseInstanceID (~> 2.0)
    - FirebaseSwizzlingUtilities (~> 1.0)
    - GoogleToolboxForMac/Logger (~> 2.1)
    - GoogleToolboxForMac/NSData+zlib (~> 2.1)
    - GTMSessionFetcher/Core (~> 1.1)
    - Protobuf (~> 3.1)
  - FirebaseSwizzlingUtilities (1.0.0)
  - GoogleToolboxForMac/Defines (2.1.3)
  - GoogleToolboxForMac/Logger (2.1.3):
    - GoogleToolboxForMac/Defines (= 2.1.3)
  - GoogleToolboxForMac/NSData+zlib (2.1.3):
    - GoogleToolboxForMac/Defines (= 2.1.3)
  - GTMSessionFetcher/Core (1.1.13)
  - nanopb (0.3.8):
    - nanopb/decode (= 0.3.8)
    - nanopb/encode (= 0.3.8)
  - nanopb/decode (0.3.8)
  - nanopb/encode (0.3.8)
  - Protobuf (3.5.0)
```

**To reproduce the crash:**
1. Install Firebase and Firebase Performance Monitoring pods and setup
2. Set `setTaskWillPerformHTTPRedirectionBlock` on `AFHTTPSessionManager`
3. Call `downloadTaskWithRequest` on `AFHTTPSessionManager` will cause the app the crash.

```
- (NSURLSessionDownloadTask *)downloadTaskWithRequest:(NSURLRequest *)request
                                             progress:(nullable void (^)(NSProgress *downloadProgress))downloadProgressBlock
                                          destination:(nullable NSURL * (^)(NSURL *targetPath, NSURLResponse *response))destination
                                    completionHandler:(nullable void (^)(NSURLResponse *response, NSURL * _Nullable filePath, NSError * _Nullable error))completionHandler;
```

Crash details:
![image](https://user-images.githubusercontent.com/3291853/36073490-a6cf754c-0f42-11e8-8abb-9ecba8a2a0ed.png)

![image](https://user-images.githubusercontent.com/3291853/36073492-ac37d344-0f42-11e8-95f4-c69e0aa2e61c.png)

The crash happens right after line 908.
![image](https://user-images.githubusercontent.com/3291853/36073543-a7fff65c-0f43-11e8-8db1-2c077d6fd820.png)

The delegate method never gets called.
![image](https://user-images.githubusercontent.com/3291853/36073550-b59b0842-0f43-11e8-9180-3150d40b0708.png)

**Comment by wimbledon** [comment]
Also reported this bug to Firebase: https://firebase.google.com/support/contact/bugs-features/?authuser=0

**Comment by jeffremer** [comment]
I received an incredibly unhelpful response from Google after filing a bug.

> Hello Jeff,

> Thanks for reaching out.

> Firebase doesn't provide support on integration with third-party libraries. You may find our list officially supported libraries here. If you're able to replicate this issue using our quickstart samples, that could get us started.

> Let me know of the results. 

> Regards,
> Gerard

**Comment by wimbledon** [comment]
I got a promising response from Google:

> Hi David,
> 
> Per checking, the status of the bug is fixed. 
> 
> As of the moment, we can't share any details or timeline for when it will be released to production. I would suggest you to check out our release notes for further updates.
> 
> If there's anything else we can help you with, just let us know. We'll try our best to help.
> 
> Regards,
> Resty

**Comment by wimbledon** [comment]
The crash fix is released as of Version 4.10.0 - March 6, 2018
https://firebase.google.com/support/release-notes/ios
